src/app/beer_garden/db/mongo/jobstore.py

```diff
@@ -24,6 +24,8 @@ def construct_trigger(trigger_type: str, bg_trigger) -> BaseTrigger:
         return IntervalTrigger(**bg_trigger.scheduler_kwargs)
     elif trigger_type == "cron":
         return CronTrigger(**bg_trigger.scheduler_kwargs)
+    elif trigger_type == "file":
+        return None
```

Comment:
I think this is causing issues? When I try to create a job with a file trigger I get the "Oh Snap" error display with this error:

> {'trigger': ["Unable to use schema. Ensure there is a deserialization_schema_selector and that it returns a schema when the function is passed in {'pattern': ['.*'], 'path': '/home/mppatri/scratch/test_files', 'recursive': False, 'callbacks': {'on_created': True, 'on_modified': False, 'on_moved': False, 'on_deleted': False}}. This is the class I got. Make sure it is a schema: None"]}

This function is supposed to construct a `BaseTrigger` of some kind so this is probably a bug?

Reply:
I've changed this to now raise an error.  The MongoDB jobstore is linked to `APScheduler` which does not support asynchronous behavior (required for `FileTrigger`).  To schedule jobs, you should be going through the top-level `MixedScheduler` class, which will schedule jobs with file triggers separately.